docs/concepts/storage/volumes.md

```diff
@@ -972,50 +971,98 @@ specification, and to select the type of media to use, for clusters that have
 
-CSI persistent volume support is an alpha feature in Kubernetes v1.9 and requires a
-cluster administrator to enable it. To enable CSI persistent volume support, the
-cluster administrator adds `CSIPersistentVolume=true` to the `--feature-gates` flag
-for apiserver, controller-manager, and kubelet.
+CSI support was introduced as alpha in Kubernetes v1.9 and moved to beta in
+Kubernets v1.10.
+
+Once a CSI compatible volume driver is deployed on a Kubernetes cluster, users
+may use the `csi` volume type to attach, mount, etc. the volumes exposed by the
+CSI driver.
+
+The `csi` volume type does not support direct reference from pod and may only be
+referenced in a pod via a `PersistentVolumeClaim` object.
 
 The following fields are available to storage administrators to configure a CSI
 persistent volume:
 
 - `driver`: A string value that specifies the name of the volume driver to use.
-  It has to be less than 63 characters and starts with a character. The driver
-  name can have '`.`', '`-`', '`_`' or digits in it.
-- `volumeHandle`: A string value that uniquely identify the volume name returned
-  from the CSI volume plugin's `CreateVolume` call. The volume handle is then
-  used in all subsequent calls to the volume driver for referencing the volume.
+  This value must corespond to the value returned in the `GetPluginInfoResponse`
+  by the CSI driver as defined in the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#getplugininfo).
+  It is used by Kubernetes to identify which CSI driver to call out to, and by
+  CSI driver components to identify which PV objects belong to the CSI driver.
+- `volumeHandle`: A string value that uniquely identifies the volume. This value
+  must correspond to the value returned in the `volume.id` field of the
+  `CreateVolumeResponse` by the CSI driver as defined in the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md#createvolume).
+  The value is passed as `volume_id` on all calls to the CSI volume driver when
+  referencing the volume.
 - `readOnly`: An optional boolean value indicating whether the volume is to be
-  published as read only. Default is false.
+  "ControllerPublished" (attached) as read only. Default is false. This value is
+  passed to the CSI driver via the `readonly` field in the
+  `ControllerPublishVolumeRequest`.
+- `fsType`: If the PV's `VolumeMode` is `Filesystem` then this field may be used
+  to specify the filesystem that should be used to mount the volume. If the
+  volume has not been formated and formating is supported, this value will be
+  used to format the volume. If a value is not specified, `ext4` is assumed.
+  This value is passed to the CSI driver via the `VolumeCapability` field of
+  `ControllerPublishVolumeRequest`, `NodeStageVolumeRequest`, and
+  `NodePublishVolumeRequest`.
+- `volumeAttributes`: A map of string to string that specifies static properties
```

Comment:
The way I read is like `volumeAttributes` are only set during ControllerCreateVolume, but in fact external provisioner can also set/use these attributes. Maybe it is worth to mention.

Reply:
They are set by external provisioner during volume creation, however if a volume is pre-provisioned and cluster admin manually creates PV objects, then the field will have to be set by cluster admin.